scripts/gallery-thumbgen.lua

```diff
@@ -35,13 +42,72 @@ function is_video(input_path)
     return false
 end
 
+function ytdl_thumbnail_url(input_path)
+    local function exec(args)
+        local ret = utils.subprocess({args = args, cancellable=false})
+        return ret.status, ret.stdout, ret
+    end
+    local function first_non_nil(x, ...)
+        if x ~= nil then return x end
+        return first_non_nil(...)
+    end
+
+    -- if input_path is youtube, generate our own URL
+    youtube_id1 = string.match(input_path, "https?://youtu.be/([%a%d%-_]+).*")
+    youtube_id2 = string.match(input_path, "https?://w?w?w?.youtube.com/v/([%a%d%-_]+).*")
+    youtube_id3 = string.match(input_path, "https?://w?w?w?.youtube.com/watch%?v=([%a%d%-_]+).*")
+    youtube_id4 = string.match(input_path, "https?://w?w?w?.youtube.com/embed/([%a%d%-_]+).*")
+    youtube_id = first_non_nil(youtube_id1, youtube_id2, youtube_id3, youtube_id4, false)
+
+    if youtube_id then
+        -- the hqdefault.jpg thumbnail should always exist, since it's used on the search result page
+        return "https://i.ytimg.com/vi/" .. youtube_id ..  "/hqdefault.jpg"
+    end
+
+    --otherwise proceed with the slower `youtube-dl -J` method
+    if not (ytdl.searched) then
+        local ytdl_mcd = mp.find_config_file("youtube-dl")
+        if not (ytdl_mcd == nil) then
+            msg.error("found youtube-dl at: " .. ytdl_mcd)
+            ytdl.path = ytdl_mcd
+        end
+        ytdl.searched = true
+    end
```

Comment:
I'm not sure about this chunk, what's the use case for it? I think it's fine to just expect `youtube-dl` to be in your `$PATH`.

Reply:
I pulled this 'searching for youtube-dl' code from ytdl_hook.lua a while ago.  My understanding is that many users (particularly windows users) won't have youtube-dl in their PATH but instead will have it sitting in their mpv config directory.   However looking at it again it appears I broke windows support in the process..

Reply:
Ah ok. From what I remember it was sufficient that ffmpeg/youtube-dl sits in the same folder as the mpv executable to be picked up by `subprocess` but if it's how ytdl_hook does it I suppose it's fine.